homeassistant/components/dlna_dms/dms.py

```diff
@@ -71,22 +69,19 @@ def __init__(
     async def async_setup_entry(self, config_entry: ConfigEntry) -> bool:
         """Create a DMS device connection from a config entry."""
         assert config_entry.unique_id
-        async with self.lock:
-            source_id = config_entry.data[CONF_SOURCE_ID]
-            assert source_id not in self.sources
-            device = DmsDeviceSource(self.hass, config_entry)
-            self.devices[config_entry.unique_id] = device
-            self.sources[device.source_id] = device
-
+        source_id = config_entry.data[CONF_SOURCE_ID]
+        assert source_id not in self.sources
```

Comment:
We only use assert in production code for type annotation to help mypy. I don't understand why we use it here.